pkg/controllers/coscheduling/podgroup_test.go

```diff
@@ -214,47 +224,56 @@ func TestFillGroupStatusOccupied(t *testing.T) {
 	}
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
-			ctrl, _, pgClient := setUp(ctx, c.podNames, c.pgName, c.podPhase, c.minMember, c.groupPhase, nil, c.podOwnerReference)
-			go ctrl.Run(1, ctx.Done())
-			err := wait.Poll(200*time.Millisecond, 1*time.Second, func() (done bool, err error) {
-				pg, err := pgClient.SchedulingV1alpha1().PodGroups("default").Get(ctx, c.pgName, metav1.GetOptions{})
-				if err != nil {
-					return false, err
-				}
-				sort.Strings(c.desiredGroupOccupied)
-				desiredGroupOccupied := strings.Join(c.desiredGroupOccupied, ",")
-				if pg.Status.OccupiedBy != desiredGroupOccupied {
-					return false, fmt.Errorf("want %v, got %v", desiredGroupOccupied, pg.Status.OccupiedBy)
-				}
-				return true, nil
+			controller, kclient := setUp(ctx, c.podNames, c.pgName, c.podPhase, c.minMember, c.groupPhase, nil, c.podOwnerReference)
+			controller.Reconcile(ctx, ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      c.pgName,
+					Namespace: metav1.NamespaceDefault,
+				},
 			})
+
+			pg := &schedulingv1a1.PodGroup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      c.pgName,
+					Namespace: metav1.NamespaceDefault,
+				},
+			}
+			err := kclient.Get(ctx, client.ObjectKeyFromObject(pg), pg)
 			if err != nil {
-				t.Fatal("Unexpected error", err)
+				t.Fatal(err)
+			}
+			sort.Strings(c.desiredGroupOccupied)
+			desiredGroupOccupied := strings.Join(c.desiredGroupOccupied, ",")
+			if pg.Status.OccupiedBy != desiredGroupOccupied {
+				t.Fatalf("want %v, got %v", desiredGroupOccupied, pg.Status.OccupiedBy)
 			}
 		})
 	}
 }
 
-func setUp(ctx context.Context, podNames []string, pgName string, podPhase v1.PodPhase, minMember int32, groupPhase v1alpha1.PodGroupPhase, podGroupCreateTime *metav1.Time, podOwnerReference []metav1.OwnerReference) (*PodGroupController, *fake.Clientset, *pgfake.Clientset) {
-	var kubeClient *fake.Clientset
-	if len(podNames) == 0 {
-		kubeClient = fake.NewSimpleClientset()
-	} else {
+func setUp(ctx context.Context,
+	podNames []string, pgName string, podPhase v1.PodPhase, minMember int32,
```

Comment:
need to format this argument

Reply:
@denkensk setUp uses the cases' options as args, there is no good way to reduce the number of args, maybe we can pass the case struct into setUp directly?

Reply:
Maybe we put an arg per line here to make it looks clear?